src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -450,14 +478,31 @@ private void deleteAllQueueItems(Message<JsonObject> event) {
         boolean unlock = payload.getBoolean(UNLOCK, false);
         String queue = payload.getString(QUEUENAME);
         redisClient.del(buildQueueKey(queue), deleteReply -> {
+            if (deleteReply.failed()) {
+                log.warn("Failed to deleteAllQueueItems.", deleteReply.cause());
+                // I like to return here. But:
+                // We need to continue below to release lock. After lock is released, impl will
+                // respond to event with ERROR anyway.
+            }
             if (unlock) {
-                redisClient.hdel(getLocksKey(), queue, unlockReply -> replyResultGreaterThanZero(event, deleteReply));
+                redisClient.hdel(getLocksKey(), queue, unlockReply -> {
+                    if (unlockReply.failed()) {
+                        log.warn("Failed to unlock queue '{}'.", queue, unlockReply.cause());
+                        replyWithStatusError(event);
+                    } else {
+                        replyResultGreaterThanZero(event, deleteReply);
+                    }
+                });
             } else {
                 replyResultGreaterThanZero(event, deleteReply);
             }
         });
     }
-    
+
+    private void replyWithStatusError(Message<?> event) {
+        event.reply("{\"status\":\"error\"}");
```

Comment:
Really use a JSON-String-Literal? Differs from all other concepts here in the code.
And: Optimizing seldom 'Error'-situations is not as good as optimizing regular occuring 'ok'-Situations.